src/postings/postings_writer.rs

```diff
@@ -155,7 +155,7 @@ pub(crate) trait PostingsWriter: Send + Sync {
     ) {
         let end_of_path_idx = term_buffer.len_bytes();
         let mut num_tokens = 0;
-        let mut end_position = 0;
+        let mut end_position = indexing_position.end_position;
```

Comment:
As I understand it, after `token_stream.process`, end_position is equivalent to

`
indexing_position.end_position + last_token.position +  last_token.position_length as u32;
`

Why do we add position_length (original token length)? The code is tested only for `position_length == 1` currently. It could be 0 or N

Reply:
let me see.

Reply:
Passing the position is done, taking in acount the token length a few lines below.

`end_position = start_position + token.position_length as u32;`

The bug we experienced was only happening in the case where there was a field value that contained no token.
(e.g empty string, just punctuation etc.)

In that case we were reinitializing the end position to 0 stupidly.
After the fix we just make the same value as the previous end position `+ POSITION_GAP`